inst/private/cygpath.m

```diff
@@ -0,0 +1,42 @@
+%% Copyright (C) 2022 Alex Vong
+%%
+%% This file is part of OctSymPy.
+%%
+%% OctSymPy is free software; you can redistribute it and/or modify
+%% it under the terms of the GNU General Public License as published
+%% by the Free Software Foundation; either version 3 of the License,
+%% or (at your option) any later version.
+%%
+%% This software is distributed in the hope that it will be useful,
+%% but WITHOUT ANY WARRANTY; without even the implied warranty
+%% of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
+%% the GNU General Public License for more details.
+%%
+%% You should have received a copy of the GNU General Public
+%% License along with this software; see the file COPYING.
+%% If not, see <https://www.gnu.org/licenses/>.
+
+%% -*- texinfo -*-
+%% @defun  cygpath ()
+%% Convert Windows native path to Cygwin POSIX-style path.
+%%
+%% @seealso{python_env_is_cygwin_like}
+%% @end defun
+
+function posix_path = cygpath (native_path)
+  %% FIXME: only allow safe characters inside "..."
+  if ~isempty (strfind (native_path, '"'))
+    error ('cygpath: native path %s must not contain "', native_path);
+  end
+
+  [status, out] = system (['cygpath -u "' native_path '"']);
+  if status ~= 0
+    error ('cygpath: cygpath exited with status %d', status);
+  end
+  assert (ischar (out));
+
+  match = regexp(out, '.*', 'match', 'dotexceptnewline');
+  assert (length (match) == 1);
+  posix_path = match{1};
```

Comment:
I'm not entirely sure what these lines are used for. But in case, you just want to strip a trailing newline, you could probably also use `strtrim`. Or am I missing something?

Reply:
I was thinking of using `strtrim` but `strtrim` removes whitespaces as well from both ends. AFAIK whitespaces are pretty common in windows path. Do you know of any function which is similar to Python `rstrip('\n')`?

Reply:
You are right: Spaces in filenames are (unfortunately) pretty common in Windows paths. But I tried to rename a file in Windows Explorer such that it ends with a space (" ") character. It looks like it is not possible to do that. That space is automatically stripped from the file name. So, using `strtrim` or `deblank` would most probably do the "correct thing". (Especially with the automatically generated file names this changeset is about.)
Using those functions might also make it a bit easier to understand what this part of the code is actually meant to be doing. (An additional comment might also help.)

Reply:
Agree, the current approach is too clumsy... Simplified and added comment as well.